### DISCUSSION TITLE
Can user get certain data or metadata from vineyard without knowing the ObjectID?

### DISCUSSION THREAD

**Discussion by Zarca** [v6d-io/v6d#252] [Q&A] [answered]
Hi,
Dose it a necessity that we must know the ObjectID of a certain object，otherwise we will not be able t get the object from vineyard server. Is that true?
Can the metadata manager be a help in such cases?

**Comment by sighingnow** [comment, marked as answer]
Without the object id how could you locate which object is that you are requiring?

**Reply by Zarca** [reply]
Is there any way to list  or search all the ObjectIDs already be put into the Vineyard?(like from etcd?)
In that way, we can then retrieve the Object with the explicit ObjectID .
I think this is very useful in the cases where 2 separate processes DO NOT have any relationships (like parent process or child process.)They can still share Objects  by searching  ObjectIDs in the etcd.

**Reply by sighingnow** [reply]
> Is there any way to list or search all the ObjectIDs already be put into the Vineyard?(like from etcd?)
> In that way, we can then retrieve the Object with the explicit ObjectID .
> I think this is very useful in the cases where 2 separate processes DO NOT have any relationships (like parent process or child process.)They can still share Objects by searching ObjectIDs in the etcd.

Yes, we have `client.ListObject()`.

**Reply by Zarca** [reply]
Thanks for that, Now I learned its use case in the test dir :) and now i've got the answer to the question above.
B.T.W, Dose the Python and C++ SDK all stored the data into an arrow::Buffer  (Which is Blob in abstract data Structure in vineyard)? If it is true, dose it means I can put a numpy into v6d through Python SDKs  and get it  as a Tensor via the C++ SDKs?

**Reply by sighingnow** [reply]
> Dose the Python and C++ SDK all stored the data into an arrow::Buffer (Which is Blob in abstract data Structure in vineyard)?

All objects are stores as a set of blobs, along with a metadata that is responsible for interpret those blobs.

> If it is true, dose it means I can put a numpy into v6d through Python SDKs and get it as a Tensor via the C++ SDKs?

Yes you can.

**Reply by Zarca** [reply]
Got it , look forward to the new release and new features of v6d :)
